crates/services/consensus_module/poa/src/service.rs

```diff
@@ -199,13 +203,17 @@ where
             predefined_blocks,
             trigger,
             sync_task_handle,
+            clock,
         }
     }
 
-    fn extract_block_info(last_block: &BlockHeader) -> (BlockHeight, Tai64, Instant) {
+    fn extract_block_info(
+        clock: &C,
+        last_block: &BlockHeader,
+    ) -> (BlockHeight, Tai64, Instant) {
         let last_timestamp = last_block.time();
         let duration_since_last_block =
-            Duration::from_secs(Tai64::now().0.saturating_sub(last_timestamp.0));
+            Duration::from_secs(clock.now().0.saturating_sub(last_timestamp.0));
         let last_block_created = Instant::now()
```

Comment:
Nope. Unfortunately, tokio time and clock time are two different things that can progress independently of each other. `clock.now()` returns a timestamp that can be communicated with the external world, whereas `Instant::now()` returns an opaque type which can only be used internally to compare against other instants.

I considered making instants a part of the `GetTime` port as well, but since tokio time is already testable (because we can run `tokio::time::pause()`, `tokio::time::advance()` etc.) I figured it would be overkill to do that.

Reply:
I see! perhaps this can be included in the follow up issue you mentioned :)

Reply:
Nice, mentioned in https://github.com/FuelLabs/fuel-core/issues/2146#issuecomment-2320487089